vpr/src/route/heap_type.h

```diff
@@ -119,6 +140,8 @@ class HeapInterface {
     // caller.
     virtual t_heap* alloc() = 0;
 
+    virtual t_heap* alloc(bool init_data_structs) = 0;
```

Comment:
Rather than adding a new virtual method for this purpose, leave the additional data structure init to the caller.

Reply:
That makes sense, and then I can do the initialization for path_data in a constructor as you suggested below.